src/providers/ModalsProvider.tsx

```diff
@@ -0,0 +1,63 @@
+import React, {
+  PropsWithChildren,
+  createContext,
+  useCallback,
+  useState,
+} from "react"
+import styles from "./ModalProvider.module.scss"
+
+interface ModalsContext {
+  content?: React.ReactNode
+  isOpen?: boolean
+  onPresent: (content: React.ReactNode, key?: string) => void
+  onClose: () => void
+}
+
+export const Context = createContext<ModalsContext>({
+  onPresent: () => undefined,
+  onClose: () => undefined,
+})
+
+const ModalsProvider: React.FC = ({ children }: PropsWithChildren<unknown>) => {
+  const [isOpen, setIsOpen] = useState(false)
+  const [content, setContent] = useState<React.ReactNode>()
+  const [, setModalKey] = useState<string>()
+
+  const handlePresent = useCallback(
+    (modalContent: React.ReactNode, key?: string) => {
+      setModalKey(key)
+      setContent(modalContent)
+      setIsOpen(true)
+    },
+    [setContent, setIsOpen, setModalKey],
+  )
+
+  const handleClose = useCallback(() => {
+    setContent(undefined)
```

Comment:
Setting to undefined can be risky as it can lead to a TypeError and crash the app. Is it possible to set content to something like `<div />` or `null` or something that empties that content without using `undefined`?